src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -394,16 +414,22 @@ object ScalafixPlugin extends AutoPlugin {
         new SemanticdbNotFound(ruleNames, scalaVersion.value, sbtVersion.value)
       ).findErrors(files, dependencies, withScalaInterface)
       if (errors.isEmpty) {
-        Def.task {
+        val task = Def.task {
+          // passively consume compilation output without triggering compile as it can result in a cyclic dependency
+          val classpath =
+            dependencyClasspath.in(config).value.map(_.data.toPath) :+
+              classDirectory.in(config).value.toPath
```

Comment:
I have left out `[copyResources`](https://github.com/sbt/sbt/blob/v1.3.13/main/src/main/scala/sbt/Defaults.scala#L3125) since I don't see why Scalafix would need it but it might be a wrong assumption.